### PR TITLE
feat(ZNTAMT-46): add support for text/xml

### PR DESCRIPTION
**Pull request by seanf**
chore: upgrade Kotlin and JaCoCo
refactor: introduce enum StringType

### CODE REVIEW

File: business-svc/src/main/java/org/zanata/magpie/util/ArticleUtil.java [hunk 1]
```diff
@@ -53,51 +56,176 @@ private ArticleUtil() {
...
+    private static int usePlaceholdersForAttributes(
+            Element document, Map<String, Node> placeholderIdMap, int prefix,
+            int counter) {
+        return usePlaceholders(placeholderIdMap, prefix, counter,
+                NON_TRANSLATABLE_ATTRIBUTE, attr ->
+                        getElementsByAttributeValue(document, attr)
+        );
+    }
+
+//    private static int usePlaceholdersForAttributes(
```
Comment (djansen-redhat) [inline business-svc/src/main/java/org/zanata/magpie/util/ArticleUtil.java]: Are we keeping this for reference?

Comment (seanf) [reply business-svc/src/main/java/org/zanata/magpie/util/ArticleUtil.java]: Not any more. Thanks.

**Comment by seanf** [comment]
@aeng are you happy with that last change (I'm not, entirely)? : https://github.com/zanata/zanata-mt/pull/85/commits/8e74fd2c544b0c5eaf2d04b60b33ec616c20801d

**Comment by aeng** [comment]
@seanf We can add in some test to increase the coverage or lower the number. I'm fine either way.

**Pull request merged**